### DISCUSSION TITLE
ESC calibration

### DISCUSSION THREAD

**Discussion by KasparsVM** [sebastiano123-c/DroneIno#2] [Q&A]
Uploaded calibration part, but I can't calibrate ESC's.
Receiver signals are strange. Throttle starts from 50% and up. Belove 50% it shifts up and down.
With the old code that used Arduino IDE setup and calibration works good, but the final part does not. 
The new code calibration does not work.

**Comment by sebastiano123-c** [comment]
Dear Kaspar,

Thank you for your report and sorry for the late reply.
The problem you describe seems strange, I never had something similar
during the development.
Maybe you can enrich your description by telling what board, IDE, ESCs and
controller you are using.
Probably something is missing in the Config.h file.
So the first thing I can encourage you to do is to read the wiki page one
more time and, in particular, how to write the Config.h file.
Please note that in the Architecture · sebastiano123-c/DroneIno Wiki ·
GitHub
<https://github.com/sebastiano123-c/DroneIno/wiki/Architecture#escs-calibration>
wiki
I suggest a video (not mine) showing how to do a proper ESC calibration.
Bests,

Seb

Il Mer 17 Mag 2023, 22:33 KasparsVM ***@***.***> ha scritto:

**Reply by KasparsVM** [reply]
Hello!
Never thought you will reply me. 

Board-ESP-WROOM-32) from AliExpress 38pin
Using PlatformIO extension in visual studio code as told in description. 
RS2205 2205 2300KV CW CCW Brushless Motors 
30A BLHeli_S ESC's
Controller is Devention DEVO 7 2.4GHz 7CH Remote with proper reciver for it, but i can't find any reciver information now on the internet. I can only provide a picture.

Setup run great. (I think. With no errors)
In calibration part in old code from the .zip file it alowed to calibrate esc's with the controller like in the video. With the main code the esc beeps as everything is and was okay. when i type r, reciver signals are mesed up. Like i told about throtle. Roll, pitch and other are mest up tooo. Gyroscope runs but also with some strange things. If i don't move the drone, One of the axes drops value every 2 or 1 sec adding -1 till it shows -45.0000 somthing somthing.

**Reply by KasparsVM** [reply]
![WhatsApp Image 2023-04-27 at 21 03 53](https://github.com/sebastiano123-c/DroneIno/assets/113163071/f80e2c93-88af-4ac5-849b-24d98afb347d)
![reciver](https://github.com/sebastiano123-c/DroneIno/assets/113163071/a7ef7dd5-6057-4622-94a5-011c48a14d08)

**Reply by KasparsVM** [reply]
![WhatsApp Image 2023-05-19 at 14 42 21](https://github.com/sebastiano123-c/DroneIno/assets/113163071/0d054c7d-6400-4ab3-9d9b-d8b7dc19c1fa)
![WhatsApp Image 2023-05-19 at 14 42 31](https://github.com/sebastiano123-c/DroneIno/assets/113163071/8772083f-fd8f-4c1e-bf59-cad783aa7dd1)
![WhatsApp Image 2023-05-19 at 14 39 32](https://github.com/sebastiano123-c/DroneIno/assets/113163071/d229d375-78b5-4ab1-a776-7c308b362d62)
![WhatsApp Image 2023-05-19 at 14 41 47](https://github.com/sebastiano123-c/DroneIno/assets/113163071/5da77627-ce44-4876-8059-68f03f2218c1)

I know. I'm using 7.4v battery, but I don't have proper 11.1v battery, but it worked with this one fine.

**Reply by KasparsVM** [reply]
7.4v battery
Gyroscope works fine in other codes. Have tested it. 
Made voltage divider for this battery and it works fine. Shows proper battery %
To power esp32 i'am using dc/dc voltage regulator and set it from 7.4v to 5.08v

**Comment by sebastiano123-c** [comment]
So, in general let me say I do not know if a 7.4V is the best solution, I
think it will probably give you power supply problems related to all the
components in the drone in the future. This is only a suggestion, and it
does not explain why the RC readings are so bad.
Another suggestion would be to check platformIO settings in the
platformio.ini. It happened sometimes that the terminal looked strange, or
gave me wrong readings.

Just to recap if I understand correctly. In the CALIBRATION mode with 7.4V
battery turned off (so esp32 is connected only to the pc) you see:
1) uncorrect RC commands;
2) uncorrect gyro readings.
Otherwise, if you connect the 7.4V battery the voltage you read on
the screen is correct?
I am correct?

If this is the case, I would proceed as follows:

1) in CALIBRATION mode, write at the end of the void setup(){} function the
following line printEEPROM();
This will let you know if after the SETUP sketch the memory has received
correct values. If eeprom shows all zeroes or other nonsense values means
the SETUP sketch failed and you have to redo the SETUP sketch.

2) If the EEPROM is ok, you can try to create a very simple sketch similar
to the SETUP sketch that only prints the bare readings of the RC controller
and the gyro. Or, you can replicate the Arduino code and find a way to
combine the two.


Il giorno ven 19 mag 2023 alle ore 13:50 KasparsVM ***@***.***>
ha scritto:

**Reply by KasparsVM** [reply]
The results in calibration are the same with or without battery when I connect esp to pc.  

Faulty receiver signals and strange things with gyro. 

About esc calibration. Do I need to calibrate each one seperatly conecting directly to reciver or when I reset the board and everything is connected as it should? 

Will test eeprom results.

**Reply by KasparsVM** [reply]
when I connect the battery it shows 10-11v in calibrations. 
I made voltage divider that would show 99% of the battery in Calibration when pressed "b".

**Comment by KasparsVM** [comment]
I can't imagine how 7.4v battery would be a problem if the ecs's are rated for 7.4 to 11.1v batterys
All the other components are using 5v.

**Comment by KasparsVM** [comment]
Sorry to bother you again.

I noticed that the old code works better for my application. 
Receiver signals are perfect, and gyroscope signals are almost perfect.
Problem starts with DroneIno folder. The last part. 

After uploading it, ESP32 reboots its self after some time. Don't know why. Can you help me with this problem. 

I did what you told me about EEPROM data and it seemed fine. Matched everything from setup, but there where still problems.
